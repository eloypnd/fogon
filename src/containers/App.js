import React, { Component } from 'react'
import { connect } from 'react-redux'
import { fetchTweets } from '../ducks/twitter'
import { Timeline, Topbar } from '../components'

export class AppContainer extends Component {
  // Lifecycle methods

  componentDidMount () {
    this.props.fetchTweets()
  }

  // render

  render () {
    const { isLoading, data } = this.props
    if (isLoading) return <div>Loading...</div>
    return (
      <React.Fragment>
        <Topbar />
        <Timeline posts={data} />
      </React.Fragment>
    )
  }
}

const mapStateToProps = state => state.twitter

const mapDispatchToProps = dispatch => ({
  fetchTweets: () => dispatch(fetchTweets())
})

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer)
