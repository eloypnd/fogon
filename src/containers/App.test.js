import React from 'react'
import renderer from 'react-test-renderer'
import { AppContainer } from './App'

it('AppContainer renders correctly', () => {
  const tree = renderer
    .create(<AppContainer fetchTweets={() => {}} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
