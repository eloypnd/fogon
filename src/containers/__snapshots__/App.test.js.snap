// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer renders correctly 1`] = `
Array [
  <div
    className="sc-Rmtcm kOUKrJ sc-bwzfXH pjOvZ"
>
    <img
        className="sc-iAyFgw jLmdZJ"
        src="logo.svg"
    />
    <h2
        className="sc-VigVT cBIDWq"
    >
        fog√≥n
    </h2>
</div>,
  <div
    className="sc-eHgmQL xPffg sc-bdVaJa iHZvIS"
/>,
]
`;
